what4/test/ExprsTest.hs

```diff
@@ -91,8 +91,66 @@ testInt = testGroup "int operators"
           diff nabs (>=) 0
         _ -> failure
   , testIntDivMod
+  , testIntMinMax
   ]
 
+testIntMinMax :: TestTree
+testIntMinMax = testGroup "int min/max"
+  [ testProperty "j <= c; c <= i -> intMax j i == intMax i j == i" $
+    property $ do
+      c <- forAll $ Gen.integral $ Range.linear (-1000) 1000
+      liftIO $ withTestSolver $ \sym -> do 
+        j <- freshBoundedInt sym (safeSymbol "j") Nothing (Just c)
+        i <- freshBoundedInt sym (safeSymbol "i") (Just c) Nothing
+        max_j_i <- intMax sym j i
+        res1 <- intEq sym max_j_i i
+        asConstantPred res1 @=? Just True
+        max_i_j <- intMax sym i j
+        res2 <- intEq sym max_i_j i
+        asConstantPred res2 @=? Just True
+  , testProperty "lo_i <= i; lo_j <= j; -> (max lo_j lo_j) <= intMax i j" $
```

Comment:
A minor thing: should `lo_j <= j; -> ...` be `lo_j <= j -> ...`? I'm not sure how to parse `; ->`.